### ISSUE TITLE
Undocumented Behavior: Not Thread-Safe in C++11

### ISSUE THREAD

**Issue by qbradq** [agauniyal/rang#53]
The implementation of rang::rang_implementation::getIword() is not thread-safe in a strictly standards-compliant C++11 compiler. According to [this](http://en.cppreference.com/w/cpp/io/ios_base/xalloc) thread-saftey for std::ios_base::xalloc was standardized in C++14. Please document the library as thread-unsafe for C++11 compilers or change the implementation to use a synchronization object when required.

**Comment by agauniyal** [comment]
This one can be an easy fix by putting a lock_guard as the first statement, but I wanted to confirm do we actually need this since the function is called to a local static initialization - 
```

	inline int getIword()
	{
		static int i = std::ios_base::xalloc();
		return i;
	}
```

**Comment by agauniyal** [comment]
we can in fact embed guards for only pre c++14 code - http://stackoverflow.com/questions/26089319/is-there-a-standard-definition-for-cplusplus-in-c14

**Comment by qbradq** [comment]
I've already got a patch for this. I noticed that you're not actually using the iword as a stream identifier (yet), only to store global state about the library. So I've replaced that method with atomic<bool>& isColorEnabled() . See PR#54

**Comment by agauniyal** [comment]
Yup that is a better method than using `std::ios_base::xalloc();` for such task. Good work 👍

**Comment by agauniyal** [comment]
solved in #54
